pkg/apiserver/handlers/webhook/convert_crd_test.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package webhook
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+var (
+	unstructuredTestData = map[string]interface{}{
+		"kind":       "kind",
+		"apiVersion": "apiVersion",
+	}
+	invalidUnstructuredTestData = map[string]interface{}{
+		"test": "test",
+	}
+
+	rawUnstructured, _ = json.Marshal(unstructuredTestData)
+	invalidRawData, _  = json.Marshal(invalidUnstructuredTestData)
+
+	testConvertFunc = func(Object *unstructured.Unstructured, version string) (*unstructured.Unstructured, metav1.Status) {
+		return &unstructured.Unstructured{}, metav1.Status{
+			Status: metav1.StatusSuccess,
+		}
+	}
+)
+
+func TestHandleCRDConversion(t *testing.T) {
+	testCases := []struct {
+		name               string
+		contentType        string
+		requestBody        interface{}
+		expectedStatusCode int
+	}{
+		{
+			name:               "Invalid content type",
+			contentType:        contentTypeHtml,
+			expectedStatusCode: http.StatusUnsupportedMediaType,
+		},
+		{
+			name:               "Invalid request body",
+			contentType:        contentTypeJson,
+			expectedStatusCode: http.StatusBadRequest,
+		},
+		{
+			name:               "Unsupported GVK",
+			contentType:        contentTypeJson,
+			expectedStatusCode: http.StatusBadRequest,
+			requestBody: &unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"kind":       "APIVersions",
+					"apiVersion": "v1",
+				},
+			},
+		},
+		{
+			name:        "Convert successfully",
+			contentType: contentTypeJson,
+			requestBody: &v1.ConversionReview{
+				TypeMeta: metav1.TypeMeta{
+					Kind:       "ConversionReview",
+					APIVersion: "apiextensions.k8s.io/v1",
+				},
+				Request: &v1.ConversionRequest{
+					UID:               "uid",
+					DesiredAPIVersion: "v1",
+					Objects: []runtime.RawExtension{
+						{
+							Raw: invalidRawData,
+						},
+					},
+				},
+			},
+		},
+		{
+			name:        "Convert failed",
```

Comment:
If the webhook runs successfully, it won't return a status code like `200` but return zero. If the convert failure, it will return zero too, but the `metav1.Status` will be `metav1.StatusFailure`. And the failure case is converged in `TestDoConversionV1`.